### ISSUE TITLE
Confidence intervals seem broader as the MATLAB version

### ISSUE THREAD

**Issue by guillermoaguilar** [wichmann-lab/python-psignifit#133]
For the data in the basic usage example, we have for the **threshold**:

`result.confidence_intervals['threshold']
`

```
[[0.0041551724137931035, 0.005110079575596817],
 [0.004218832891246685, 0.005110079575596817],
 [0.004346153846153846, 0.004982758620689655]]

```

and in MATLAB:

95 % = 0.004282936763270,  0.004964252219404
90% =  0.004340714759236, 0.004907784893587
68 % = 0.004455989969640,  0.004793927224722

This is noticeable when plotting the marginals, that's how I found it. 
In python for 95% threshold:

![marginal_threshold_python](https://github.com/user-attachments/assets/682e61d0-e27e-48de-963c-6838ace6acae)

In MATLAB for 95% threshold:

![marginal_threshold_MATLAB](https://github.com/user-attachments/assets/d0e06196-7944-44ee-b50e-7930c8e3825e)

In python the upper bound is >0.005, in MATLAB is <0.005.

**Comment by guillermoaguilar** [comment]
Same for **width**

Python, 95 % confidence: [0.00300887573964497, 0.006772189349112426]

![marginal_width_python](https://github.com/user-attachments/assets/86a66142-15ca-4428-a48e-3a16a04fc908)



MATLAB, 95 % confidence:   0.003532170611945   0.006040905831117

![marginal_width_MATLAB](https://github.com/user-attachments/assets/bc6dafdd-b387-49c0-8b6f-7fc0ace04407)

**Comment by dekuenstle** [comment]
The good news is that apparently, the marginal is computed correctly: Both curves match exactly if overlayed. 
![overlay_ci](https://github.com/user-attachments/assets/11f26618-191a-4400-9a2d-c9cd88ed77b0)

@guillermoaguilar For debugging it would be helpful to get more details. Which CI calculation method did you use in both MATLAB and Python? [This package](https://github.com/wichmann-lab/python-psignifit/blob/main/psignifit/_confidence.py) supports both `project` (default) and `percentile`.

**Comment by dekuenstle** [comment]
There still is a systematic difference between MATLAB and Python CIs as discussed [in the MATLAB test PR](https://github.com/wichmann-lab/python-psignifit/pull/66#issuecomment-2366811726). 

> The python CI still remain systematically smaller (~5%). In my perspective this is not a bug: While the MATLAB sets the border in the middle of two grid points (one within, one outside the CI), while the Python version interpolates the cumulative distribution. In steep regimes like the one below, the interpolation must be narrower.

@guillermoaguilar @otizonaizit  @pberkes  @lschwetlick  What is your opinion, bug or feature?

**Comment by guillermoaguilar** [comment]
I ran the fits above with *defaults*, default in MATLAB and default in python. Now running it with 'percentiles' it gives me

`result.confidence_intervals['threshold']`

```
[[0.004251168834040448, 0.00493232186010108],
 [0.00430891627918812, 0.004875907476743896],
 [0.004424169210878712, 0.004762087037380426]]
```

which is much closer to MATLAB:

95 % = 0.004282936763270, 0.004964252219404
90% = 0.004340714759236, 0.004907784893587
68 % = 0.004455989969640, 0.004793927224722

So **differences are due to the discrepency in the default method to compute the CI*.
IMO we *must* change the default to 'percentiles', and all defaults should be all the same w.r.t. the MATLAB implementation. 

I see that David @dekuenstle updated the PR #66 with the change, but I think this change has to be done ASAP and in isolation. 

@otizonaizit do you agree?

**Comment by guillermoaguilar** [comment]
> There still is a systematic difference between MATLAB and Python CIs as discussed [in the MATLAB test PR](https://github.com/wichmann-lab/python-psignifit/pull/66#issuecomment-2366811726).
> 
> > The python CI still remain systematically smaller (~5%). In my perspective this is not a bug: While the MATLAB sets the border in the middle of two grid points (one within, one outside the CI), while the Python version interpolates the cumulative distribution. In steep regimes like the one below, the interpolation must be narrower.
> 
> @guillermoaguilar @otizonaizit @pberkes @lschwetlick What is your opinion, bug or feature?

I think it boils down to  coverage. If the narrower python CI have still appropriate coverage, then these should be kept

Felix said in the presentation of the paper that coverage was sometimes too large than expected  (Fig. 6 C and D in the paper), so this might be the reason. But one would have to actually run the simulations to establish coverage for this python implementation.

**Comment by otizonaizit** [comment]
> There still is a systematic difference between MATLAB and Python CIs as discussed [in the MATLAB test PR](https://github.com/wichmann-lab/python-psignifit/pull/66#issuecomment-2366811726).
> 
> > The python CI still remain systematically smaller (~5%). In my perspective this is not a bug: While the MATLAB sets the border in the middle of two grid points (one within, one outside the CI), while the Python version interpolates the cumulative distribution. In steep regimes like the one below, the interpolation must be narrower.
> 
> @guillermoaguilar @otizonaizit @pberkes @lschwetlick What is your opinion, bug or feature?

I think this is a question for @FelixWichmann , even if my gut feeling is that 5% is small enough that it doesn't really matter.

**Comment by otizonaizit** [comment]
> I ran the fits above with _defaults_, default in MATLAB and default in python. Now running it with 'percentiles' it gives me
> 
> `result.confidence_intervals['threshold']`
> 
> ```
> [[0.004251168834040448, 0.00493232186010108],
>  [0.00430891627918812, 0.004875907476743896],
>  [0.004424169210878712, 0.004762087037380426]]
> ```
> 
> which is much closer to MATLAB:
> 
> 95 % = 0.004282936763270, 0.004964252219404 90% = 0.004340714759236, 0.004907784893587 68 % = 0.004455989969640, 0.004793927224722
> 
> So differences **are due** to the discrepancy in the default method to compute the CI. IMO we _must_ change the default to 'percentiles', and all defaults should be all the same w.r.t. the MATLAB implementation.
> 
> I see that David @dekuenstle updated the PR #66 with the change, but I think this change has to be done ASAP and in isolation. Users will wonder why CI are sufficiently different between implementations.
> 
> @otizonaizit do you agree?

Yes, I have the feeling that it would be better to have a separate PR just to change the default mode for CI calculation, merge the obvious PRs that are already there and do a quick 4.2 release.

What is @FelixWichmann thinking about this? Which method should be default?

**Comment by otizonaizit** [comment]
@guillermoaguilar : Felix approved the change in an email exchange.

I changed the default CI method in #135 . Could you wait for #132 to be merged and then cut a 4.2 release? In two weeks we can discuss about the further steps, but this fix is urgent enough that it warrants a release right now, I think.
